aws/cloudformation/standalone/gen_ai_curriculum/deploy_gen_ai_curriculum_stack

```diff
@@ -74,7 +70,9 @@ def main
     print "\nFetching HuggingFace Access Token..."
     secrets_manager = Aws::SecretsManager::Client.new
     value = secrets_manager.get_secret_value(
-      {secret_id: "#{options[:environment]}/gen-ai-curriculum/huggingface_token"}
+      # Use the development secret since this script is running locally
+      # and we don't have permissions to access production/test secrets.
+      {secret_id: "development/cdo/gen_ai_curriculum_huggingface_token"}
```

Comment:
@cat5inthecradle let me know how you feel about this - if I try to access a `production/` or `test/` secret, I get `Error User: arn:aws:sts::475661607190:assumed-role/Engineering_FullAccess/sanchit@code.org is not authorized to perform: secretsmanager:GetSecretValue on resource: production/cdo/gen_ai_curriculum_huggingface_token with an explicit deny in an identity-based policy`.

I'm guessing that permissions to access these secrets are limited to production and test execution roles?

Reply:
I think we should stick to environment specific secrets as a security standard. We can solve this by moving the secret lookup.

The cloudformation stack is being deployed with the CloudformationService IAM Role, which does have permission to fetch the production secret. [So we can move the secret lookup inside the template](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references-secretsmanager.html).

```
      PrimaryContainer:
        Environment:
          {
            "HF_MODEL_ID": "<%= config[:hf_model_id] %>",
            "HUGGING_FACE_HUB_TOKEN": !Sub "{{resolve:ssm-secure:/${Environment}/cdo/gen_ai_curriculum_huggingface_token}}",
            "MAX_INPUT_LENGTH": 3000,
            "MAX_TOTAL_TOKENS": 4096,
            "SAGEMAKER_CONTAINER_LOG_LEVEL": 20,
            "SAGEMAKER_REGION": !Sub "${AWS::Region}",
            "SM_NUM_GPUS": 1,
          }
        Image: !Sub "${HuggingFaceImageUri}"
        Mode: "SingleModel"
```

However... I'm curious, would anything other than this cloudformation template need to access the huggingface token secret?

Reply:
Right now, I don't believe so - maybe if we had other Huggingface models we wanted to deploy for other AI products down the road?